src/crimson/os/seastore/cached_extent.h

```diff
@@ -567,6 +577,7 @@ class ExtentIndex {
 
   void erase(CachedExtent &extent) {
     assert(extent.parent_index);
+    assert(extent.is_linked());
     auto erased = extent_index.erase(
```

Comment:
I think I have noticed the same issue before, and this is why I added line 585 - 587 below in the commit https://github.com/ceph/ceph/commit/3c8d12217e6efc8f660f26fb7e01faf0223b131a in order to get the correct metric value, as I observed that `erase()` appears not always successful. But that time the program won't break due to futile erase operations.

I think this is a cleaner fix that prevents futile erases, so should we instead `assert(erased)` below?

Reply:
yes